libparsec/src/config.rs

```diff
@@ -38,6 +44,8 @@ impl From<ClientConfig> for Arc<libparsec_client::ClientConfig> {
     }
 }
 
+const DEFAULT_PREVENT_SYNC_PATTERN: &str = std::include_str!("default_pattern.ignore");
```

Comment:
I'd rather avoid having the default prevent sync pattern in the `libparsec_client` crate.

The issue is `libparsec_client` is high level stuff without tests (it is expected that GUI + CLI relying on it are enough to detect the issues)

On the other hand, the default prevent sync pattern is most likely something we want to test !

So I think what we should do:
- move the default prevent sync pattern into the `libparsec_types` crate
- Add a `Regex::default()` method that build the regex from the default prevent sync pattern

This way it is simple to add tests in `libparsec_types` checking the behavior of the default prevent sync pattern (and compare it to the empty and custom patterns ^^), and building `ClientConfig` is trivial:
```rust
            prevent_sync_pattern: match config.prevent_sync_pattern_file.unwrap_or_default()
```

(this works this way given the bindings code should introduce a new `Regex` type which is `StrBasedType` and hence does the string validation in the javascript-to-rust conversion step)